MergerServiceUnitTests/Utils/TaskUtilsTest.cs

```diff
@@ -59,20 +65,82 @@ public void WhenGettingTaskByType_ShouldUseCorrectHttpUri()
     }
 
+    public static IEnumerable<object[]> GetBadJsonTestParameters()
+    {
+        yield return new object[] { "bad json" };
+        yield return new object[] { File.ReadAllText("invalidTask.json") };
+        yield return new object[] { File.ReadAllText("invalidTaskStrategy.json") };
+    }
+
     [TestMethod]
-    public void WhenGettingMalformedJsonTask_ShouldReturnNull()
+    [DynamicData(nameof(GetBadJsonTestParameters), DynamicDataSourceType.Method)]
+    public void WhenGettingMalformedJsonTask_ShouldReturnNull(string json)
     {
-      var malformedJsonString = "bad json";
-      this._httpClientMock.Setup(httpClient => httpClient.PostData(It.IsAny<string>(), It.IsAny<HttpContent?>(), It.IsAny<bool>())).Returns(malformedJsonString);
+      this._httpClientMock.Setup(httpClient => httpClient.PostData(It.IsAny<string>(), It.IsAny<HttpContent?>(), It.IsAny<bool>())).Returns(json);
 
       var testTaskUtils = new TaskUtils(_configurationManagerMock.Object, _httpClientMock.Object, _taskUtilsLoggerMock.Object,
         _testActivitySource);
 
+      Console.WriteLine(json);
+
       var resultTask = testTaskUtils.GetTask("testJobType", "testTaskType");
 
       Assert.IsNull(resultTask);
     }
 
+    public static IEnumerable<object[]> GetGoodJsonTestParameters()
+    {
+        yield return new object[] { File.ReadAllText("validTaskNoStrategy.json") };
+        yield return new object[] { File.ReadAllText("validTaskMixedStrategy.json") };
+    }
+
+    [TestMethod]
+    [DynamicData(nameof(GetGoodJsonTestParameters), DynamicDataSourceType.Method)]
+    public void WhenGettingJsonTask_ShouldReturnTaskObject(string json)
+    {
+      int maxAttempts = this._configurationManagerMock.Object.GetConfiguration<int>("TASK", "maxAttempts");
+      this._httpClientMock.Setup(httpClient => httpClient.PostData(It.IsAny<string>(), It.IsAny<HttpContent?>(), It.IsAny<bool>())).Returns(json);
+
+      var testTaskUtils = new TaskUtils(_configurationManagerMock.Object, _httpClientMock.Object, _taskUtilsLoggerMock.Object,
+        _testActivitySource);
+      var resultTask = testTaskUtils.GetTask("testJobType", "testTaskType");
+
+      Assert.IsNotNull(resultTask);
+      
+      // Validate GUIDs
+      Assert.IsTrue(Guid.TryParse(resultTask.Id, out _));
+      Assert.IsTrue(Guid.TryParse(resultTask.JobId, out _));
+
+      // Validate dates
+      Assert.IsTrue(DateTime.TryParse(resultTask.Created.ToString(), out _));
+      Assert.IsTrue(DateTime.TryParse(resultTask.Updated.ToString(), out _));
+
+      Assert.IsNotNull(resultTask.Parameters);
+      Assert.IsTrue(resultTask.Parameters.Sources?.Length > 0);
+      Assert.IsTrue(resultTask.Parameters.Batches?.Length > 0);
+      Assert.IsInstanceOfType(resultTask.Parameters.TargetFormat, typeof(TileFormat));
+      Assert.IsInstanceOfType(resultTask.Parameters.OutputFormatStrategy, typeof(FormatStrategy));
```

Comment:
why don't you assert the FormatStrategy value?

Reply:
It doesn't matter as long as it's compatible with the `enum`.